src/rgw/rgw_multi.cc

```diff
@@ -302,12 +322,17 @@ int list_bucket_multiparts(rgw::sal::RGWRadosStore *store, RGWBucketInfo& bucket
   list_op.params.ns = RGW_OBJ_NS_MULTIPART;
   list_op.params.filter = &mp_filter;
 
-  return(list_op.list_objects(max_uploads, objs, common_prefixes, is_truncated, null_yield));
+  return(list_op.list_objects(max_uploads, objs, common_prefixes, is_truncated, null_yield, this_parent_span));
 }
 
 int abort_bucket_multiparts(rgw::sal::RGWRadosStore *store, CephContext *cct, RGWBucketInfo& bucket_info,
-				string& prefix, string& delim)
+				string& prefix, string& delim, const Span& parent_span)
 {
+  char buffer[1000];
+  get_span_name(buffer , __FILENAME__,  "function",   __PRETTY_FUNCTION__);
+  Span span_1 = child_span(buffer, parent_span);
+  const Span& this_parent_span(span_1);
```

Comment:
@suab321321 what is the difference between this_parent_span and parent_span, iirc you specified earlier, but seems like conversation is lost somewhere

Reply:
funcA(parent_span){
Span this_parent_span;
B(this_parent_span);
}
parent_span is for child span created in A but for the functions called inside A, those function will use this_parent_span as theri parent_span

Reply:
you can use propagation span/local/global parent span; i think this doesn't catch this idea quite well